firebase-database/src/androidTest/java/com/google/firebase/database/QueryTest.java

```diff
@@ -4556,22 +4558,31 @@ public void testGetReturnsNullForEmptyNodeWhenOnline()
   }
 
   @Test
-  public void testGetThrowsExceptionForEmptyNodeWhenOffline()
-      throws DatabaseException, InterruptedException {
+  public void testGetWaitsForConnection() throws DatabaseException, InterruptedException {
     FirebaseApp app =
         appForDatabaseUrl(IntegrationTestValues.getNamespace(), UUID.randomUUID().toString());
     FirebaseDatabase db = FirebaseDatabase.getInstance(app);
-    DatabaseReference node = db.getReference().child(db.getReference().push().getKey());
+    DatabaseReference node =
+        db.getReference().child(Objects.requireNonNull(db.getReference().push().getKey()));
     db.goOffline();
+    AtomicBoolean receivedValue = new AtomicBoolean();
+    THREAD_POOL_EXECUTOR.execute(
+        () -> {
+          try {
+            Thread.sleep(3000L);
```

Comment:
We can now reduce this sleep to something much lower in order not to slow down our tests.